ocaml/xapi/helpers.ml

```diff
@@ -1220,15 +1220,9 @@ let local_storage_exists () =
   with _ -> false
 
 let touch_file fname =
-  try
-    if fname <> "" then
-      match Unixext.spawnvp "touch" [|"touch"; fname|] with
-      | Unix.WEXITED 0 ->
-          ()
-      | _ ->
-          warn "Unable to touch ready file '%s': touch exited abnormally" fname
-  with e ->
-    warn "Unable to touch ready file '%s': %s" fname (Printexc.to_string e)
+  if fname <> "" then
```

Comment:
What is the point of this test? I would rather see the log entry that this can't be touched than nothing.

Reply:
This function gets used on values set by the configuration. These values use the convention of using empty to mean None, as in "please don't try to create the file". I think adding an error to the logs would be confusing in that case (which is the default for some values)

Reply:
So we keep it here but this sounds to me like something that should be handled on the calling side and not here or at the very least needs to be made explicit with an option.

Reply:
Stdlib's Arg module doesn't support using `string option ref`: it forces to process string parameters with the empty string being the null value. I've made it obvious this function processes args, but I don't think we can do much more short of changing the framework for parsing args

Reply:
Fair enough. `Arg` is convenient but I would favour `Cmdliner` for new code because it is more compositional and has richer functionality.